tracer/src/Datadog.Trace/Agent/DiscoveryService/DiscoveryService.cs

```diff
@@ -42,10 +44,18 @@ private DiscoveryService(IApiRequestFactory apiRequestFactory)
 
         public string AgentVersion { get; private set; }
 
-        public static DiscoveryService Create(IApiRequestFactory apiRequestFactory)
+        public static DiscoveryService Create(ImmutableExporterSettings exporterSettings)
         {
             lock (GlobalLock)
             {
```

Comment:
Is there likely to actually be any contention around this? I assume not (we probably have an issue if there is), so this doesn't really matter, but currently we always create a new `apiRequestFactory` even if we don't need it, and take the lock unnecessarily. Probably can be ignored given this should only be called once (currently)

Reply:
It should be called only once, and currently called only once 😄